Add custom errorLabel to MaterialTextField

ItoroD opened the pull request:
I replaced helpLabel with error label where necessary. Tested validation and it works correctly. This should avoid any bind errors
@HenrikJannsen 
- #1783

HenrikJannsen reviewed (changes requested):
NACK
If the help text is used and the validation triggers an error the error is displayed above the help.

The `// TODO add custom errorLabel...` can be removed.
At `onWidthChanged` the `errorLabel.setPrefWidth(width - 2 * errorLabel.getLayoutX());` is missing

Best to check the `reformat` checkbox at the commit interface in IntelliJ so that code is correctly formatted (just a missing space where you add the errorLabel)

ItoroD commented:
@HenrikJannsen I have made the required changes. It was best to set the help to invisible when there is an error message and visible when no error. Other option was making help empty but that made the components below move up so that was not a good option.


![Screenshot 2024-03-23 094215](https://github.com/bisq-network/bisq2/assets/17571526/3c54a521-e493-4c33-b518-cfd9f2f60268)

HenrikJannsen commented:
> @HenrikJannsen I have made the required changes. It was best to set the help to invisible when there is an error message and visible when no error. Other option was making help empty but that made the components below move up so that was not a good option.
> 
> ![Screenshot 2024-03-23 094215](https://private-user-images.githubusercontent.com/17571526/316233461-3c54a521-e493-4c33-b518-cfd9f2f60268.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExODkzNjgsIm5iZiI6MTcxMTE4OTA2OCwicGF0aCI6Ii8xNzU3MTUyNi8zMTYyMzM0NjEtM2M1NGE1MjEtZTQ5My00YzMzLWI1MTgtY2ZkOWYyZjYwMjY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIzVDEwMTc0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkMmEwY2NkYTJjZTRjNGQyYzUwMmU5ZDFmNGQyZTEzMWFjY2JiNTY0YjBhMDUxOGM1NjIyYjYyNzU0N2ZkZjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4-yOj7rRPEC1X07xRu0nw_eD1meIC1kpde6bmiFHDBA)

Does the error message replace the help text? So if the error message gets cleared the help text is back?

ItoroD commented:
> > @HenrikJannsen I have made the required changes. It was best to set the help to invisible when there is an error message and visible when no error. Other option was making help empty but that made the components below move up so that was not a good option.
> > ![Screenshot 2024-03-23 094215](https://private-user-images.githubusercontent.com/17571526/316233461-3c54a521-e493-4c33-b518-cfd9f2f60268.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExODkzNjgsIm5iZiI6MTcxMTE4OTA2OCwicGF0aCI6Ii8xNzU3MTUyNi8zMTYyMzM0NjEtM2M1NGE1MjEtZTQ5My00YzMzLWI1MTgtY2ZkOWYyZjYwMjY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIzVDEwMTc0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkMmEwY2NkYTJjZTRjNGQyYzUwMmU5ZDFmNGQyZTEzMWFjY2JiNTY0YjBhMDUxOGM1NjIyYjYyNzU0N2ZkZjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4-yOj7rRPEC1X07xRu0nw_eD1meIC1kpde6bmiFHDBA)
> 
> Does the error message replace the help text? So if the error message gets cleared the help text is back?

Yes. When error is cleared help message is back

ItoroD commented:
https://recordit.co/pOJNoUPsde

@HenrikJannsen

ItoroD commented:
@HenrikJannsen 

any other thing I need to do here? Is this good for merge? 
I shall move to another issue.

HenrikJannsen commented:
The height is not updated if there is no help text. I guess the update() method calls are missing.

<img width="473" alt="Screenshot 2024-03-24 at 12 25 21" src="https://github.com/bisq-network/bisq2/assets/116298498/5c6680eb-9780-4866-8fbf-e480db7a9b98">

HenrikJannsen commented on apps/desktop/desktop/src/main/java/bisq/desktop/components/controls/MaterialTextField.java:
```diff
@@ -186,18 +186,22 @@ public void setValidators(ValidatorBase... validators) {
         getActiveValidator().ifPresentOrElse(validator -> {
                     errorLabel.setText(validator.getMessage());
                     errorLabel.pseudoClassStateChanged(PSEUDO_CLASS_ERROR, true);
+                    helpLabel.setOpacity(0);
```
Better use setManaged(false) and setVisible(false) as that takes the component out of the render cycle and is better for performance. We use opacity only for dimming/fade effect and in some rare cases where setManaged causes too much troubles.

ItoroD replied:
Yes suggestion makes sense for optimization. Note that If i use setManaged(false) and setVisible(false) here, when you click out with mouse update is triggered which sets the help text to visible again making error message and helptext overlap

Update():

    helpLabel.setVisible(StringUtils.isNotEmpty(helpProperty().get()));
    helpLabel.setManaged(StringUtils.isNotEmpty(helpProperty().get()));
    
hence, this will mean making StringUtils.isNotEmpty(helpProperty().get()) return false when there is supposed to be an error message so that on update we do not make helpLabel visible when the error state is not clear yet

ItoroD replied:
And for isNotEmpty(helpProperty().get()) to be false, we will have to  set helpLabel.setText("") something like:

validate(): 

  ```
getActiveValidator().ifPresentOrElse(validator -> {
                      errorLabel.setText(validator.getMessage());
                      errorLabel.pseudoClassStateChanged(PSEUDO_CLASS_ERROR, true);
                      helpLabel.setManaged(false);
                      helpLabel.setVisible(false);
                      helpLabel.setText("");
                  },
                  () -> {
                      errorLabel.setText("");
                      errorLabel.pseudoClassStateChanged(PSEUDO_CLASS_ERROR, false);
                      helpLabel.setManaged(true);
                      helpLabel.setVisible(true);
                      helpLabel.setText(globalHelpText);
                  });
```

which means we have to store the help text somewhere first

`String globalHelpText =  "";`

MaterialTextField(@Nullable String description, @Nullable String prompt, @Nullable String help, @Nullable String value): 

```
if (StringUtils.isNotEmpty(help)) {
            helpLabel.setText(help);
            globalHelpText = help;
        }
```

I did this before but figured it was a bit long. If you think this is ok then I will do it

HenrikJannsen reviewed (changes requested):
NACK

ItoroD commented:
> The height is not updated if there is no help text. I guess the update() method calls are missing.
> 
> <img alt="Screenshot 2024-03-24 at 12 25 21" width="473" src="https://private-user-images.githubusercontent.com/116298498/316285328-5c6680eb-9780-4866-8fbf-e480db7a9b98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEyODEzOTEsIm5iZiI6MTcxMTI4MTA5MSwicGF0aCI6Ii8xMTYyOTg0OTgvMzE2Mjg1MzI4LTVjNjY4MGViLTk3ODAtNDg2Ni04ZmJmLWU0ODBkYjdhOWI5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyNFQxMTUxMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNzcxMGZjY2ZjYzdlNWEzZTQ1NWUzNzYwNjYzZWU4ZjUyYWM2YTA2ZThhNjEwZmE5YTIwMWJiNTVkZDQ1Y2M5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jCCwqWYozAB__PULE6MAxJvvCsi2tqtkVD7HmxrMGQg">

No this is because this is a Vbox component and spacing between them have been set to 10. So it does not care whether we have an error to display or not because the error component is not part of the first Vbox component like help text is  (for places we have help)

`VBox networkVBox = new VBox(10, difficultyAdjustmentFactor, ignoreDiffAdjustFromSecManagerSwitch);`

when i change the spacing: 

`VBox networkVBox = new VBox(20, difficultyAdjustmentFactor, ignoreDiffAdjustFromSecManagerSwitch);`

![image](https://github.com/bisq-network/bisq2/assets/17571526/befff9c2-458b-47bc-8b79-edd0f2b4bd3c)

My suggestion is simple 
- for places where there is no help text and component is vbox, we give Vbox spaceing of 20 to accommodate for when there is error 
- for places where there is help text if we use the `helpLabel.setManaged(false); helpLabel.setVisible(false);` like you suggested, when we show error we will have the same spacing issue as you have pointed out. Because vbox for component with help also have spacing 10. 
`VBox tradeVBox = new VBox(10, minRequiredReputationScore, offersOnlySwitch, closeMyOfferWhenTaken);`
 
If we use opacity to make help not visible we maintain the correct spacing for error to display (by using helps space)

ItoroD commented:
Similar thing already exist for backup location spacing already 20
```

backupButton = new Button(Res.get("settings.utils.backup.backupButton"));
HBox backupButtons = new HBox(10, setBackupLocationButton, backupButton);
VBox backupBox = new VBox(20, backupLocation, backupButtons);
```

ItoroD commented:
> > The height is not updated if there is no help text. I guess the update() method calls are missing.
> > <img alt="Screenshot 2024-03-24 at 12 25 21" width="473" src="https://private-user-images.githubusercontent.com/116298498/316285328-5c6680eb-9780-4866-8fbf-e480db7a9b98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEyODEzOTEsIm5iZiI6MTcxMTI4MTA5MSwicGF0aCI6Ii8xMTYyOTg0OTgvMzE2Mjg1MzI4LTVjNjY4MGViLTk3ODAtNDg2Ni04ZmJmLWU0ODBkYjdhOWI5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyNFQxMTUxMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNzcxMGZjY2ZjYzdlNWEzZTQ1NWUzNzYwNjYzZWU4ZjUyYWM2YTA2ZThhNjEwZmE5YTIwMWJiNTVkZDQ1Y2M5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jCCwqWYozAB__PULE6MAxJvvCsi2tqtkVD7HmxrMGQg">
> 
> No this is because this is a Vbox component and spacing between them have been set to 10. So it does not care whether we have an error to display or not because the error component is not part of the first Vbox component like help text is (for places we have help)
> 
> `VBox networkVBox = new VBox(10, difficultyAdjustmentFactor, ignoreDiffAdjustFromSecManagerSwitch);`
> 
> when i change the spacing:
> 
> `VBox networkVBox = new VBox(20, difficultyAdjustmentFactor, ignoreDiffAdjustFromSecManagerSwitch);`
> 
> ![image](https://private-user-images.githubusercontent.com/17571526/316306323-befff9c2-458b-47bc-8b79-edd0f2b4bd3c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEyODU3MTYsIm5iZiI6MTcxMTI4NTQxNiwicGF0aCI6Ii8xNzU3MTUyNi8zMTYzMDYzMjMtYmVmZmY5YzItNDU4Yi00N2JjLThiNzktZWRkMGYyYjRiZDNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI0VDEzMDMzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlZGViZjVjMzNlNTJlOThkN2I1YTg5NDEwMmExNGI0ZDM4MTIwNmI1Njc4ZDY3MzUyOTYzOWY3Yjg2ODU4ODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Fdpx7tMKjSRqSpFyoTQM6kRsZDgxF8lLIp_sy6mT4pU)
> 
> My suggestion is simple
> 
> * for places where there is no help text and component is vbox, we give Vbox spaceing of 20 to accommodate for when there is error
> * for places where there is help text if we use the `helpLabel.setManaged(false); helpLabel.setVisible(false);` like you suggested, when we show error we will have the same spacing issue as you have pointed out. Because vbox for component with help also have spacing 10.
>   `VBox tradeVBox = new VBox(10, minRequiredReputationScore, offersOnlySwitch, closeMyOfferWhenTaken);`
> 
> If we use opacity to make help not visible we maintain the correct spacing for error to display (by using helps space)


```
backupButton = new Button(Res.get("settings.utils.backup.backupButton"));
HBox backupButtons = new HBox(10, setBackupLocationButton, backupButton);
VBox backupBox = new VBox(20, backupLocation, backupButtons);
```

ItoroD commented:
@HenrikJannsen updated the code

HenrikJannsen commented:
I am busy atm. Will review a bit later.

ItoroD commented:
> I am busy atm. Will review a bit later. Anyone else can review as well.

ok

HenrikJannsen reviewed (changes requested):
NACK

This is not the appropriate change to make the component generic. We do not want to adjust the outside container but the height calculation should take care of the potential existence of the error or help field. With the error field getting visible the height should gets updated so that the outside container re-layouts and maintains the intended spacing. 

I will not review personally more PR's as my time would be better spent by fixing it myself. Any other contributor though is welcome to review upcoming PRs. Just wanted to give you my feedback so that you can manage expectations.

ItoroD commented:
Noted @HenrikJannsen

ItoroD commented:
@HenrikJannsen I know you have been very patient with me and this PR in taking the time to look at it as you have other pressing issues to look at. Only now do I see what you mean by height calculation should take of the error text showing. Should have seen  this earlier (my bad) last commit

- uses setManaged and setVisible instead of setopacity
- added height calculation for error text label. 
- reverted changed height of vbox to what it was before

if you have not fixed this already and theres a 1% chance of taking a look at this, it will be great as I am now getting familiar with the code base.

axpoems reviewed (changes requested):
NACK


The following comment is not addressed with the latest changes. Help text is not getting back:

> Does the error message replace the help text? So if the error message gets cleared the help text is back?

axpoems commented on apps/desktop/desktop/src/main/java/bisq/desktop/components/controls/MaterialTextField.java:
```diff
@@ -529,7 +560,9 @@ protected double getFieldLayoutY() {
         if (helpLabel.isManaged()) {
             return helpLabel.getLayoutY() + helpLabel.getHeight();
-        } else {
+        } else if(errorLabel.isManaged()){
+            return errorLabel.getLayoutY() + errorLabel.getHeight();
+        }else{
```
This was not addressed either:

> Best to check the reformat checkbox at the commit interface in IntelliJ so that code is correctly formatted (just a missing space where you add the errorLabel)

ItoroD commented:
> NACK
> 
> The following comment is not addressed with the latest changes. Help text is not getting back:
> 
> > Does the error message replace the help text? So if the error message gets cleared the help text is back?

Hi @axpoems help text does come back for places that had help text. Not all fields have help text except if there is another part of the app I am not seeing this

axpoems commented:
![errLabel](https://github.com/bisq-network/bisq2/assets/145597137/e75387d3-7815-4dc5-83d5-1454aa917819)

Once I clear I would expect to see the help text.

ItoroD commented:
@axpoems This is what I have. And I have pushed this

https://recordit.co/r5Qjsv3THF

ItoroD commented:
the field cannot be left empty as empty is also an error. it has to be a number placed in there

ItoroD commented:
> ![errLabel](https://private-user-images.githubusercontent.com/145597137/318180087-e75387d3-7815-4dc5-83d5-1454aa917819.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3ODgwMDIsIm5iZiI6MTcxMTc4NzcwMiwicGF0aCI6Ii8xNDU1OTcxMzcvMzE4MTgwMDg3LWU3NTM4N2QzLTc4MTUtNGRjNS04M2Q1LTE0NTRhYTkxNzgxOS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMzMFQwODM1MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MmFlYjc0OGNmNjA3ZjQ3Zjg3ZGUzNTQzMGRlNjM1ODViYzg1NGQzNzAwMzNkMTY3ZGFiMTA0ZjE4ZWQyZmQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qCh6V-r95ygvfgxe_XqTlcsH8yp-7SFxxDjnexCR-ck) [ ![errLabel](https://private-user-images.githubusercontent.com/145597137/318180087-e75387d3-7815-4dc5-83d5-1454aa917819.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3ODgwMDIsIm5iZiI6MTcxMTc4NzcwMiwicGF0aCI6Ii8xNDU1OTcxMzcvMzE4MTgwMDg3LWU3NTM4N2QzLTc4MTUtNGRjNS04M2Q1LTE0NTRhYTkxNzgxOS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMzMFQwODM1MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MmFlYjc0OGNmNjA3ZjQ3Zjg3ZGUzNTQzMGRlNjM1ODViYzg1NGQzNzAwMzNkMTY3ZGFiMTA0ZjE4ZWQyZmQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qCh6V-r95ygvfgxe_XqTlcsH8yp-7SFxxDjnexCR-ck) ](https://private-user-images.githubusercontent.com/145597137/318180087-e75387d3-7815-4dc5-83d5-1454aa917819.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3ODgwMDIsIm5iZiI6MTcxMTc4NzcwMiwicGF0aCI6Ii8xNDU1OTcxMzcvMzE4MTgwMDg3LWU3NTM4N2QzLTc4MTUtNGRjNS04M2Q1LTE0NTRhYTkxNzgxOS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMzMFQwODM1MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MmFlYjc0OGNmNjA3ZjQ3Zjg3ZGUzNTQzMGRlNjM1ODViYzg1NGQzNzAwMzNkMTY3ZGFiMTA0ZjE4ZWQyZmQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qCh6V-r95ygvfgxe_XqTlcsH8yp-7SFxxDjnexCR-ck) [ ](https://private-user-images.githubusercontent.com/145597137/318180087-e75387d3-7815-4dc5-83d5-1454aa917819.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3ODgwMDIsIm5iZiI6MTcxMTc4NzcwMiwicGF0aCI6Ii8xNDU1OTcxMzcvMzE4MTgwMDg3LWU3NTM4N2QzLTc4MTUtNGRjNS04M2Q1LTE0NTRhYTkxNzgxOS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMzMFQwODM1MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MmFlYjc0OGNmNjA3ZjQ3Zjg3ZGUzNTQzMGRlNjM1ODViYzg1NGQzNzAwMzNkMTY3ZGFiMTA0ZjE4ZWQyZmQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qCh6V-r95ygvfgxe_XqTlcsH8yp-7SFxxDjnexCR-ck)
> 
> Once I clear I would expect to see the help text.

So until there is a number placed in the field error will not go away. So even if it empty error does not go "Must be Number"

ItoroD commented:
I have enabled reformat code now.

djing-chan commented:
@ItoroD Is the PR ready for review?
e.g.
- Tested with all the use cases (with and without help text)
- Changing the height when the error text comes in the no help text use case and removes the error text once the validator has no error

E.g.  In case the PR is not ready for review (e.g. under dev) please convert it to a draft PR.

ItoroD commented:
@djing-chan yes it is ready for review.

 I was hoping @axpoems will retest has he made a mistake in testing. Appropriate value should be put in field before error clears for help text to come back. 

Height changes now when error shows

djing-chan commented:
@ItoroD 
It is not adjusting the height:

<img width="575" alt="Screenshot 2024-04-01 at 13 20 04" src="https://github.com/bisq-network/bisq2/assets/146422224/9d0e84f1-4eb1-451d-a4f0-aa8914c9c52b">

<img width="514" alt="Screenshot 2024-04-01 at 13 20 08" src="https://github.com/bisq-network/bisq2/assets/146422224/7cfa0ade-c4e3-470c-ba6d-8e83a33a70a1">

When the error message is not visible the overall height of the component should be adopted to the height needed.

ItoroD commented:
@djing-chan 
revert resize was missing after resize

https://github.com/bisq-network/bisq2/assets/17571526/442152ae-4a17-4cd2-bf95-fdd64922858f

djing-chan commented:
> @djing-chan revert resize was missing after resize
> 
>  short.mp4

Looks ok now from the video.
Can you squash all your commits and rebase to main?
I will give it a final review afterwards.

ItoroD commented:
@djing-chan 

rebase done.

djing-chan commented:
Squashing all commits on top of the rebase would be good. You can select your commits in IntelliJ in the Github commit history and squash them. Then run a rebase on main.

ItoroD commented:
this better? @djing-chan

HenrikJannsen reviewed (approved):
ACK

Rebase was still not ok. As this PR takes too much effort with reviews I made an own PR here:
https://github.com/bisq-network/bisq2/pull/1992

And a follow up PR with further improvements at https://github.com/bisq-network/bisq2/pull/1991

You still can make a compensation request for your work included in that PR.
